modules/core/src/main/java/org/apache/ignite/internal/GridJobExecuteResponse.java

```diff
@@ -219,147 +264,82 @@ public boolean retry() {
      * @return Topology version for that specified partitions haven't been reserved
      *          on the affinity node.
      */
-    public AffinityTopologyVersion getRetryTopologyVersion() {
-        return retry != null ? retry : AffinityTopologyVersion.NONE;
+    public @Nullable AffinityTopologyVersion retryTopologyVersion() {
+        return retry;
```

Comment:
Is null check is still necessary? If so, let't perform check in place where `retryTopologyVersion` is invoked.